packages/syft/src/syft/custom_worker/runner_k8s.py

```diff
@@ -198,12 +263,14 @@ def _create_stateful_set(
                             "containers": [
                                 {
                                     "name": pool_name,
+                                    "imagePullPolicy": "IfNotPresent",
                                     "image": tag,
                                     "env": env_clone,
                                     "volumeMounts": [creds_volume_mount],
                                 }
                             ],
                             "volumes": [creds_volume],
+                            "imagePullSecrets": pull_secret_obj,
```

Comment:
During my testing, I found that the deployment fails if the imagePullSecrets is None

We could either set it to null or 
omit completey
![Screenshot 2024-02-02 at 11 05 14 AM](https://github.com/OpenMined/PySyft/assets/43314053/888a82ca-bc9a-4fad-b656-30c9f3da64a5)
Sample run setting the value as None

Reply:
I did the testing on trying to do helm install by setting imagepullsecrets as None in backend-statefulset.yml

Reply:
Wroker pod spec isn't created & applied until we call the API. This definitely has to do with the static YAML files.

Reply:
Also you're setting it as `None` which is a string in YAML, it should be `null`. Python `None` maps to YAML `null` automatically.